samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -64,21 +65,25 @@
   private String newJobModelVersion;  // version published in ZK (by the leader)
   private JobModel jobModel;
 
-  public ZkJobCoordinator(String processorId, String groupId, Config config, ScheduleAfterDebounceTime debounceTimer, ZkUtils zkUtils,
+  public ZkJobCoordinator(String groupId, Config config, ScheduleAfterDebounceTime debounceTimer,
                           SamzaContainerController containerController) {
-    this.processorId = processorId;
-    this.zkUtils = zkUtils;
-    this.keyBuilder = zkUtils.getKeyBuilder();
     this.debounceTimer = debounceTimer;
     this.containerController = containerController;
-    this.zkController = new ZkControllerImpl(processorId, zkUtils, debounceTimer, this);
     this.config = config;
+    this.processorId = new ApplicationConfig(config).getProcessorId();
```

Comment:
You shouldn't need this if you are using ZkJobCoordinatorFactory as it is defined.